wiki/development/coding-guidelines.md

```diff
@@ -380,33 +382,80 @@ If a function returns error information, then that error information will be tes
 There shall be no unreachable code.
 
 ### 6.3. Function Parameters
-Parameters of functions must be retrieved through the usage of the param or params commands. If the function is part of the public API, parameters must be checked on allowed data types and values through the usage of the param and params commands.
+Parameters of functions must be retrieved through the usage of the `param` or `params` commands. If the function is part of the public API, parameters must be checked on allowed data types and values through the usage of the param and params commands.
 
 Usage of the CBA Macro `PARAM_x` or `BIS_fnc_Param` is deprecated and not allowed within the ACE project unless very specific reasons allow no other alternative.
 
 ### 6.4. Return Values
-Functions and code blocks that specific a return a value must have a meaningfull return value.
+Functions and code blocks that specific a return a value must have a meaningfull return value. If no meaningfull return value, the function should return nil.
 
 ### 6.5. Private Variables
-All private variables shall be declared in a private array.
+All private variables shall make use of the `private` keyword on initalization. When declaring a private variable before initalization, usage of the private array syntax is allowed. All private variables must be either initialized using the private key word, or declared using the private array syntax. Exceptions to this rule are variables obtained from an array. Note that this may only be down by making use of the `params` command family, as this ensures the variable is declared as private.
+
+Good:
+```js
+private _myVariable = "hello world";
+```
+
+Good:
+```js
+_myArray params ["_elementOne", "_elementTwo"];
+```
 
+Bad:
 ```js
-private ["_myVariable"];
+_elementOne = _myArray select 0;
+_elementTwo = _myArray select 1;
 ```
 
+
 ### 6.6. Lines of Code
 Any one function shall contain no more than 250 lines of code, excluding the function header and any includes.
 
 ### 6.7. Variable declarations
 Declarations should be at the smallest feasible scope.
 
+
+Good:
+```js
+if (call FUNC(myCondition)) then {
+   private _areAllAboveTen = true; // <- smallest feasable scope
+   
+   {
+      if (_x >= 10) then {
+         _areAllAboveTen = false;
+      };
+   } forEach _anArray;
+   
+   if (_areAllAboveTen) then {
+       hint "all values are above ten!";
+   };
+}
+```
+
+Bad:
+```sqf
+private _areAllAboveTen = true; // <- this is bad, because it can be initalized in the if statement
+if (call FUNC(myCondition) then {
```

Comment:
I didn't think it was worth to mention, but this line is also missing the same closing parenthesis.